pitop/labs/webserver/static/joysticks.js

```diff
@@ -0,0 +1,106 @@
+var cmd_vel_twist = {
+  'linear': {
+    'x': 0.0
+  },
+  'angular': {
+    'z': 0.0
+  }
+};
+
+var pan_tilt_twist = {
+  'angular': {
+    'y': 0.0,
+    'z': 0.0
+  }
+};
+
+const MAX_LINEAR_SPEED = 0.44;
+const MAX_ANGULAR_SPEED = 5.12;
+const MAX_SERVO_SPEED = 100;
+var publishCmdVelImmediately = true;
+var publishPanTiltImmediately = true;
+
+function cmdVelTwistPublisher(linear, angular) {
+    if (linear !== undefined && angular !== undefined) {
+        cmd_vel_twist.linear.x = linear;
+        cmd_vel_twist.angular.z = angular;
+    } else {
+        cmd_vel_twist.linear.x = 0;
+        cmd_vel_twist.angular.z = 0;
+    }
+    // console.log(linear)
+    window.command['cmd_vel'](cmd_vel_twist);
+}
+
+function panTiltTwistPublisher(angular_y, angular_z) {
+    if (angular_y !== undefined && angular_z !== undefined) {
+        pan_tilt_twist.angular.y = angular_y;
+        pan_tilt_twist.angular.z = angular_z;
+    } else {
+        pan_tilt_twist.linear.x = 0;
```

Comment:
there's a copy-paste bug here, pan_tilt_twist doesn't have a linear component. However, the new implementation of setting the angle directly instead of the speed actually works quite well, so see other comments